src/test/ui/closures/issue-84128.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-84128.rs:13:13
+   |
+LL |         Foo(())
+   |             ^^ expected integer, found `()`
+   |
+note: return type inferred to be `{integer}` here
```

Comment:
It's not really the 'return type' that's being inferred here

Reply:
I missed that, thanks. I will fix it.

Reply:
Shouldn't it either be "return type inferred to be `Foo<{integer}>` here" with span of `Foo(0)` indicated; or else "`{integer}` inferred here" (with the span of `0` indicated)?  (My personal view, FWIW, is that the former is a tad more informative but either seems fine).

As it is, it's a bit confusing: essentially "`{integer}` inferred here" but `Foo(0)` indicated.